controllers/expose/portHandler.go

```diff
@@ -0,0 +1,278 @@
+	}
+
+	for _, port := range exposedPorts {
+		lbService.Spec.Ports = append(lbService.Spec.Ports, getServicePortFromExposedPort(targetServiceName, port))
+	}
+
+	return lbService, true
+}
+
+// filterTargetServicePorts returns all ports from the service filtered by the service name prefix.
+// If the service has no ports in the service it additionally returns false. Otherwise, true.
+func filterTargetServicePorts(targetServiceName string, lbService *corev1.Service) ([]corev1.ServicePort, bool) {
+	var servicePorts []corev1.ServicePort
+	found := false
+
+	for _, servicePort := range lbService.Spec.Ports {
+		servicePortName := servicePort.Name
+		servicePrefix := getTargetServicePortNamePrefix(targetServiceName)
+		f := strings.HasPrefix(servicePortName, servicePrefix)
+		if !f {
+			servicePorts = append(servicePorts, servicePort)
+		} else {
+			found = true
+		}
+	}
+
+	return servicePorts, found
+}
+
+func (eph *exposedPortHandler) updateService(ctx context.Context, exposedService *corev1.Service) error {
+	_, err := eph.serviceInterface.Update(ctx, exposedService, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update %s service: %w", cesLoadbalancerName, err)
+	}
+	return nil
+}
+
+// RemoveExposedPorts removes given dogu exposed ports from the loadbalancer service.
+// If these ports are the only ones, the service will be deleted.
+// If the dogu has no exposed ports, the method returns nil.
+func (eph *exposedPortHandler) RemoveExposedPorts(ctx context.Context, serviceName string) error {
+	logger := log.FromContext(ctx)
+
+	logger.Info("Delete exposed tcp and upd ports...")
+	err := eph.ingressController.DeleteExposedPorts(ctx, eph.namespace, serviceName)
+	if err != nil {
+		return fmt.Errorf("failed to delete entries from expose configmap: %w", err)
+	}
+
+	retryErr := retry.OnConflict(func() error {
+		exposedService, err := eph.getCesLoadBalancerService(ctx)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return fmt.Errorf("failed to get service %s: %w", cesLoadbalancerName, err)
+			} else {
+				return nil
+			}
+		}
+
+		ports, found := filterTargetServicePorts(serviceName, exposedService)
+		if !found {
+			logger.Info(fmt.Sprintf("found no exposed ports for service %s in loadbalancer service %s", serviceName, cesLoadbalancerName))
+			return nil
+		}
+
+		logger.Info("Update loadbalancer service...")
+		// Do not delete the loadbalancer service even it has no ports because this could result in a new ip if it gets recreated.
+		exposedService.Spec.Ports = ports
+		return eph.updateService(ctx, exposedService)
+	})
+
+	if retryErr != nil {
+		return fmt.Errorf("failed to remove exposed ports from loadbalancer service %s: %w", cesLoadbalancerName, retryErr)
+	}
+
+	return nil
+}
+
+func equalsServicePortExposedPort(servicePort corev1.ServicePort, exposedPort util.ExposedPort) bool {
+	if strings.ToUpper(string(servicePort.Protocol)) != strings.ToUpper(string(exposedPort.Protocol)) {
```

Comment:
🚫 **[checkstyle]** <staticcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA6005: should use strings.EqualFold instead
<!-- __reviewdog__:ChA5NGY4MTA2ODc0MTZiMGYwEgpjaGVja3N0eWxl -->